docs/reference/components/sinks/elasticsearch.cue

```diff
@@ -236,9 +246,19 @@ components: sinks: elasticsearch: {
 			title: "Conflicts"
 			body: """
 				Vector [batches](#buffers--batches) data flushes it to Elasticsearch's
-				[`_bulk` API endpoint][urls.elasticsearch_bulk]. All events are inserted
+				[`_bulk` API endpoint][urls.elasticsearch_bulk]. By default, all events are inserted
 				via the `index` action. In the case of an conflict, such as a document with the
-				same `id`, Vector will add or _replace_ the document as necessary.
+				same `id`, Vector will add or _replace_ the document as necessary. If `bulk_action` is
```

Comment:
Is it Vector that will add or replace or Elasticsearch?

With `index`, if an action entry already has an `_id` field specified, won't Elasticsearch perform an "update" (causing segment merges due to the implicit delete that happens)?  And if an action entry does _not_ have an `_id` field, won't Elasticsearch will generate a unique `_id` value itself, creating a new document in the target index?

So with the `index` action there is really no notion of "conflict".

The notion of a conflict only comes with the `create` action and the action entry has an `_id` field specified.  In that case, if a document exists with the same `_id`, Elasticsearch returns a 409 conflict, else the new document is added.  If the `create` action is used without an `_id` field specified, no conflict will occur because a unique ID will be assigned by Elasticsearch and inserted.

Reply:
If you specify an `_id` and "index" a document if there is an existing document with that `_id` it should be replaced upon indexing. Copying from the ES docs:

```
index

    (Optional, string) Indexes the specified document. If the document exists, replaces the document and increments the version.
```

I believe the original documentation is accurate if considering the "conflict" being the same as replacing an existing document.

Reply:
Hopefully we have somehow communicated our definition of "conflict", which is different from how Elasticsearch uses it (conflict only mentioned with "create").

And it would be great to make sure the wording here places the ownership of the behaviors taken on Elasticsearch and not on Vector (replace "Vector" with "Elasticsearch" in "Vector will add or _replace_ the document as necessary").

Reply:
@binarylogic let me know your thoughts on this - I had intentionally left the original descriptions "as is" as possible.

Reply:
I tend to agree that "conflict" is a bad word for what happens for `index` actions. I'd reword this as:

```
Vector [batches](#buffers--batches) data flushes it to Elasticsearch's [`_bulk` API endpoint][urls.elasticsearch_bulk]. By default, all events are inserted via the `index` action which will update documents if an existing one has the same `id`. If `bulk_action` is configured with `create`, Elasticsearch will _not_ replace an existing document and instead return a conflict error.
```